### ISSUE TITLE
visual editor: markdown table style varies across document open/save cycles

### ISSUE THREAD

**Issue by jjallaire** [rstudio/rstudio#8671]
From @mine-cetinkaya-rundel

I have a file I’m editing with the visual editor (in bookdown, if that is relevant) where I have two tables and the formatting of the tables seem to change sporadically where sometimes the lines between rows are gone (see here) and sometimes they’re back (see here). I can’t put my finger on what I might be doing to trigger this, any suggestions?

https://github.com/rstudio-education/datascience-box/commit/4945c504d6b9a60f49e2ddc6b4d612ce01a64581#diff-6f5e342191d8b85f06e823eacc34471c9adfd677611ddd37ea9900934950d0c6
https://github.com/rstudio-education/datascience-box/commit/a06e66eb01d2dcd26917b2fd84e603a8eb172c86#diff-6f5e342191d8b85f06e823eacc34471c9adfd677611ddd37ea9900934950d0c6

In case helpful, I feel like this happens when the table column widths change, and they seem to sometimes change when I reopen a file.

From @jjallaire 

I think that's it. Grid tables are used for explicitly sized widths, and when you re-open a file depending on the width of the editor pane sometimes "explicitly sized" gets triggered when it shouldn't

**Comment by jjallaire** [comment]
@mine-cetinkaya-rundel In my testing I can get the table to go from pipe to grid as a result of the editor width changing but once it goes to grid it stays that way. Have you definitely seen it go from grid to pipe?

**Comment by jjallaire** [comment]
It looks to me like saving from source mode (w/ `canonical: true`) might be this what explains the switching back and forth. It looks like visual mode prefers grid tables esp. when the editor is sized narrowly, while source mode seems to prefer pipe tables. Could this explain the variation you have been seeing?

**Comment by mine-cetinkaya-rundel** [comment]
Yes, I think it does explain it @jjallaire. I was switching back and forth between source and vidual editor to try to figure out what might be happening to the tables.

**Comment by jjallaire** [comment]
Okay, a quick first attempt to repro based on saving from source mode didn't yield any changes between table formats but I'll keep experimenting. Thanks!

**Comment by mine-cetinkaya-rundel** [comment]
I was working on a paper with tables with looong text in cells where I observed this back and forth behaviour again. I tried to capture what I'm seeing in a smaller example here: https://github.com/mine-cetinkaya-rundel/viz-editor-tables. Perhaps the diffs in the commits there will be helpful? I'm just guessing here but maybe it has to do with how much text is in table and/or column widths ever having been resized?

**Comment by mrajeev08** [comment]
This happens to me as well (changing tables from pipe to grid but not the switching back and forth)! 

It would be nice if the table format remained unchanged from the source if toggling between visual and source.  I mainly use the visual markdown editor for the citation functionality, so if  https://github.com/rstudio/rstudio/issues/7876 is fixed then that would do the trick for me 😉 

Thanks so much for the great IDE!

**Comment by mine-cetinkaya-rundel** [comment]
This is still an issue that should be addressed.

**Comment by bleds22e** [comment]
Agreed!

**Comment by austinwpearce** [comment]
I think this is still an issue.

**Comment by mine-cetinkaya-rundel** [comment]
This is still true.
Version 2025.08.0-daily+286 (2025.08.0-daily+286)

**Comment by bleds22e** [comment]
Can confirm! It is very frustrating while teaching...
